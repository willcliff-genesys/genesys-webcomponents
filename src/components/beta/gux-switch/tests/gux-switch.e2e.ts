import { newE2EPage } from '@stencil/core/testing';
import '../../../../components';

describe('gux-switch', () => {
  const html = `
  <gux-switch-beta lang-"en" layout="small" value="day">
    <gux-switch-item value="month">Month</gux-switch-item>
    <gux-switch-item value="week">Week</gux-switch-item>
    <gux-switch-item value="day">Day</gux-switch-item>
    <gux-switch-item value="hour" disabled>Hour</gux-switch-item>
    <gux-switch-item value="minute">Minute</gux-switch-item>
  </gux-switch-beta>
`;
  describe('#render', () => {
    it(`should render as expected`, async () => {
      const page = await newE2EPage({ html });
      const element = await page.find('gux-switch-beta');

      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('#interactions', () => {
    it(`should change value on gux-switch-item click`, async () => {
      const page = await newE2EPage({ html });
      const element = await page.find('gux-switch-beta');
      const guxSwitchItemMinute = await page.find(
        'gux-switch-item[value=minute]'
      );

      expect(await element.getProperty('value')).toBe('day');

      guxSwitchItemMinute.click();
      await page.waitForChanges();

      expect(await element.getProperty('value')).toBe(
        await guxSwitchItemMinute.getProperty('value')
      );
    });

    it(`should not change value on gux-switch-item click if it is disabled`, async () => {
      const page = await newE2EPage({ html });
      const element = await page.find('gux-switch-beta');
      const guxSwitchItemHour = await page.find('gux-switch-item[value=hour]');
      const currentValue = await element.getProperty('value');

      expect(currentValue).toBe('day');

      guxSwitchItemHour.click();
      await page.waitForChanges();

      expect(await element.getProperty('value')).toBe(currentValue);
    });

    it(`should emit a 'change' and 'input' event when a new item is selected`, async () => {
      const page = await newE2EPage({ html });
      const element = await page.find('gux-switch-beta');
      const guxSwitchItemMinute = await page.find(
        'gux-switch-item[value=minute]'
      );

      const spyOnInputEvent = await element.spyOnEvent('input');
      const spyOnChangeEvent = await element.spyOnEvent('change');

      guxSwitchItemMinute.click();
      await page.waitForChanges();

      expect(spyOnInputEvent.length).toBe(1);
      expect(spyOnChangeEvent.length).toBe(1);
    });

    it(`should not emit a 'change' or 'input' event when a disabled item is selected`, async () => {
      const page = await newE2EPage({ html });
      const element = await page.find('gux-switch-beta');
      const guxSwitchItemHour = await page.find('gux-switch-item[value=hour]');

      const spyOnInputEvent = await element.spyOnEvent('input');
      const spyOnChangeEvent = await element.spyOnEvent('change');

      guxSwitchItemHour.click();
      await page.waitForChanges();

      expect(spyOnInputEvent.length).toBe(0);
      expect(spyOnChangeEvent.length).toBe(0);
    });
  });
});
