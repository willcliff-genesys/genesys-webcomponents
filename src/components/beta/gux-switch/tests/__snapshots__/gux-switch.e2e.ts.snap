// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-switch #render should render as expected 1`] = `"<gux-switch-beta lang-\\"en\\"=\\"\\" layout=\\"small\\" value=\\"day\\" role=\\"group\\" class=\\"gux-small hydrated\\"><!----> <gux-switch-item value=\\"month\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-switch-item\\">Month</button></gux-switch-item> <gux-switch-item value=\\"week\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-switch-item\\">Week</button></gux-switch-item> <gux-switch-item value=\\"day\\" class=\\"gux-selected hydrated\\"><!----><button type=\\"button\\" class=\\"gux-switch-item\\">Day</button></gux-switch-item> <gux-switch-item value=\\"hour\\" disabled=\\"\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-switch-item\\" disabled=\\"\\">Hour</button></gux-switch-item> <gux-switch-item value=\\"minute\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-switch-item\\">Minute</button></gux-switch-item> </gux-switch-beta>"`;
